src/components/backend-ai-registry-list.ts

```diff
@@ -373,101 +395,115 @@ class BackendAIRegistryList extends BackendAIPage {
+   */
+  private _launchDialogById(id) {
     this.shadowRoot.querySelector(id).show();
   }
 
-  _hideDialogById(id) {
+  /**
+   * Hide dialog by element id
+   * 
+   * @param {string} id - element id starts from `#`
+   */
+  private _hideDialogById(id) {
     this.shadowRoot.querySelector(id).hide();
   }
 
-  _openCreateRegistryDialog() {
-    this.editMode = false;
-    this.selectedIndex = -1;
-    this.registryType = 'docker';
-    this._launchDialogById('#add-registry-dialog');
+  /**
+   * Open create registry dialog with initial value
+   * NOTE: Initial registry type is `docker`.
+   */
+  private _openCreateRegistryDialog() {
+    this._editMode = false;
+    this._selectedIndex = -1;
+    this._registryType = 'docker';
+    this.requestUpdate();  // call for explicit update
+    this._launchDialogById('#configure-registry-dialog');
   }
 
-  _hideValidationMessage() {
-    this.shadowRoot.querySelector('#registry-hostname-validation').style.display = 'none';
-    this.shadowRoot.querySelector('#registry-url-validation').style.display = 'none';
-    this.shadowRoot.querySelector('#project-name-validation').style.display = 'none';
+  /**
+   * Reset validation message in registry configuation dialog
+   */
+  private _resetValidationMessage() {
+    /**
+     * FIXME: need to change repetitive value manipulation in future.
+     */
+    this._registryHostnameValidationMsg.style.display = 'none';
+    this._registryUrlValidationMsg.style.display = 'none';
+    this._projectNameValidationMsg.style.display = 'none';
   }
 
-  _openEditRegistryDialog(registry) {
-    this.editMode = true;
+  /**
+   * Open registry configuration dialog by hostname
+   * 
+   * @param hostname 
+   */
+  private _openEditRegistryDialog(hostname) {
+    this._editMode = true;
     let registryInfo;
-    for (let i = 0; i < this.registryList.length; i++) {
-      if (this.registryList[i].hostname === registry) {
-        registryInfo = this.registryList[i];
+    for (let i = 0; i < this._registryList.length; i++) {
+      if (this._registryList[i].hostname === hostname) {
+        registryInfo = this._registryList[i];
         break;
       }
     }
     if (!registryInfo) {
-      globalThis.notification.show(`No such registry: ${registry}`);
+      globalThis.notification.show(`No such registry hostname: ${hostname}`);
       return;
     }
-    this.registryList[this.selectedIndex] = registryInfo;
-    this._hideValidationMessage();
-    this._launchDialogById('#add-registry-dialog');
-  }
-
-  _toggleProjectNameInput() {
-    this.registryType = this.shadowRoot.querySelector('#select-registry-type').value;
-    this._validateProjectName();
+    this._registryList[this._selectedIndex] = registryInfo;
+    this._registryType = this._registryList[this._selectedIndex]?.type as string;
+    this.requestUpdate();  // call for explicit update
+    this._resetValidationMessage();
+    this._launchDialogById('#configure-registry-dialog');
   }
 
```

Comment:
The function doesn't do anything for validation but just toggles the visibility of the message of corresponding input field. So I changed the function name to a more precise name.